File: golang/153-find-minimum-in-rotated-sorted-array.go
package golang

import "math"

func findMin(nums []int) int {
    lo, hi := 0, len(nums) - 1
    for lo < hi {
        if nums[lo] < nums[hi] {
            break
        }
        mid := int(math.Floor(float64((lo + hi) / 2)))
        if nums[lo] <= nums[mid] {
            lo = mid + 1
        } else {
            hi = mid
        }
    }
    return nums[lo]
}

File: javascript/153-find-minimum-in-rotated-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
 var findMin = function(nums) {    
    let lo = 0, hi = nums.length-1
    while (lo < hi) {
        if (nums[lo] < nums[hi]) break  
        const mid = Math.floor((lo + hi)/2)        
        if (nums[lo] <= nums[mid]) {
            lo = mid + 1
        } else {
            hi = mid
        }
    }
    return nums[lo]
};